src/GraphQL/ParseGraphQLSchema.js

```diff
@@ -8,18 +8,32 @@ import * as parseClassQueries from './loaders/parseClassQueries';
 import * as parseClassMutations from './loaders/parseClassMutations';
 import * as defaultGraphQLQueries from './loaders/defaultGraphQLQueries';
 import * as defaultGraphQLMutations from './loaders/defaultGraphQLMutations';
+import GraphQLController from '../Controllers/GraphQLController';
+import DatabaseController from '../Controllers/DatabaseController';
 
 class ParseGraphQLSchema {
-  constructor(databaseController, log) {
+  databaseController: DatabaseController;
+  graphQLController: GraphQLController;
+
+  constructor(params: {
+    databaseController: DatabaseController,
+    graphQLController: GraphQLController,
+    log: any,
+  }) {
+    this.graphQLController =
+      params.graphQLController ||
+      requiredParameter('You must provide a graphQLController instance!');
     this.databaseController =
-      databaseController ||
+      params.databaseController ||
       requiredParameter('You must provide a databaseController instance!');
-    this.log = log || requiredParameter('You must provide a log instance!');
+    this.log =
+      params.log || requiredParameter('You must provide a log instance!');
   }
 
   async load() {
-    const schemaController = await this.databaseController.loadSchema();
-    const parseClasses = await schemaController.getAllClasses();
+    await this._initializeSchemaAndConfig();
+
+    const parseClasses = await this._getClassesForSchema();
     const parseClassesString = JSON.stringify(parseClasses);
 
     if (this.graphQLSchema) {
```

Comment:
I think that the verification in the line 40 will not work anymore unless we save the original parseClass and graphQLConfig that came from the controllers and check both.